lumispy/utils/analysis.py

```diff
@@ -0,0 +1,117 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019-2022 The LumiSpy developers
+#
+# This file is part of LumiSpy.
+#
+# LumiSpy is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the license, or
+# (at your option) any later version.
+#
+# LumiSpy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with LumiSpy. If not, see <https://www.gnu.org/licenses/#GPL>.
+
+import hyperspy.api as hs
+
+
+def plot_span_map(sig, nspans=1):
+    """
+    Plot a span map.
+    
+    Plots a navigator consisting of the sum over all positions in sig, with `nspans` spans. 
+
+    For each span a `BaseSignal` image is plotted consisting of integrated counts over the the range defined by the span.
+
+    The spans can be moved interactively and the corresponding images will update automatically.
+
+    Arguments
+    ---------
+    sig: hyperspy.signals.Signal1D
+        hyperspectra to inspect. Should have 2 spatial dimensions and 1 signal dimension.
+    nspans: int
+        number of channels to plot. Maximum is 3, default is 1.
+
+    Returns
+    -------
+    all_sum: hs.signals.BaseSignal
+        Sum over all positions of `sig`, the 'navigator' for `plot_span_map`
+    spans: hs.roi.SpanROI
+        The span ROIs from the navigator
+    span_sigs: hs.signals.BaseSignal
+        slices of `sig` according to each span roi
+    span_sums: hs.signals.BaseSignal
+        the summed `span_sigs`.
+    """
+    if nspans > 3:
+        raise ValueError("Maximum number of spans allowed is 3")
+
+    if (
+        sig.axes_manager.signal_dimension != 1
+        or sig.axes_manager.navigation_dimension != 2
+    ):
+        sig_dims, nav_dims = (
+            sig.axes_manager.signal_dimension,
+            sig.axes_manager.navigation_dimension,
+        )
+        raise ValueError(
+            f"This method is designed for data with 1 signal and 2 navigation dimensions, not {sig_dims} and {nav_dims} respectively"
```

Comment:
I think you could easily also make it work for a 1 signal 1 nav axis with little work.

Reply:
I'm trying to get my head around what this would look like.

We have a line of spectra, the 'nav' is the sum over these spectra, then the signal is a line showing the intensity at each point.

tbf this might work already, just have to remove the type check!... and write the tests.

I'd just be a bit worried that then the docstring is would start to get quite confusing - is it still `plot_span_map`, or is that more `plot_span_line`!?

This is maybe an argument for adding this as a method to the various signal types defined in lumispy.

Reply:
Indeed, that is the linescans I mentioned above. You select regions in the spectrum and then get a lineplot of the intensity against position. There would be a difference in the plotting, I guess, as it is not a map any more.

The signal types in lumispy can all have 0, 1 or 2 navigation dimensions. So a differentiation does not make sense. Only, time-resolved data has in theory an additional signal axis, which complicates things (but in the long term it would be cool to also mark regions along the time trace and get the maps for that).

Reply:
I agree. I would definitely make sure it works for 0, 1 and 2 dimensional navigation axis (0 returns single value, 1 returns a line plot, 2 returns an image).
As for more than 1 dimensional signal axes, I think it goes beyond this PR.

Reply:
I have nothing against having different functions for 0NavD, 1NavD, 2NavD signals but I let you decide :)